lightning/src/chain/keysinterface.rs

```diff
@@ -182,21 +187,58 @@ pub trait ChannelKeys : Send {
 /// A simple implementation of ChannelKeys that just keeps the private keys in memory.
 pub struct InMemoryChannelKeys {
 	/// Private key of anchor tx
-	pub funding_key: SecretKey,
+	funding_key: SecretKey,
 	/// Local secret key for blinded revocation pubkey
-	pub revocation_base_key: SecretKey,
+	revocation_base_key: SecretKey,
 	/// Local secret key used in commitment tx htlc outputs
-	pub payment_base_key: SecretKey,
+	payment_base_key: SecretKey,
 	/// Local secret key used in HTLC tx
-	pub delayed_payment_base_key: SecretKey,
+	delayed_payment_base_key: SecretKey,
 	/// Local htlc secret key used in commitment tx htlc outputs
-	pub htlc_base_key: SecretKey,
+	htlc_base_key: SecretKey,
 	/// Commitment seed
-	pub commitment_seed: [u8; 32],
+	commitment_seed: [u8; 32],
+	/// Local public keys and basepoints
+	pub(crate) local_channel_pubkeys: ChannelPublicKeys,
 	/// Remote public keys and base points
-	pub remote_channel_pubkeys: Option<ChannelPublicKeys>,
+	pub(crate) remote_channel_pubkeys: Option<ChannelPublicKeys>,
 	/// The total value of this channel
-	pub channel_value_satoshis: u64,
+	channel_value_satoshis: u64,
+}
+
+impl InMemoryChannelKeys {
+	/// Create a new InMemoryChannelKeys
+	pub fn new<C: Signing>(
+		secp_ctx: &Secp256k1<C>,
+		funding_key: SecretKey,
+		revocation_base_key: SecretKey,
+		payment_base_key: SecretKey,
+		delayed_payment_base_key: SecretKey,
+		htlc_base_key: SecretKey,
+		commitment_seed: [u8; 32],
+		channel_value_satoshis: u64,
+		remote_channel_pubkeys: Option<ChannelPublicKeys>) -> InMemoryChannelKeys {
```

Comment:
Hmm, this is awkward - its pub and if a user ever sets it to Some() for any reason we'll almost certainly panic() in set_remote_channel_pubkeys pretty quick. You should be able to just bypass the new() call in the deserialize and construct manually.

Reply:
Ah, you are right.  Fixed.